sdks/python/apache_beam/ml/inference/huggingface_inference.py

```diff
@@ -638,8 +640,30 @@ def __init__(
     if max_batch_size is not None:
       self._batching_kwargs['max_batch_size'] = max_batch_size
     self._large_model = large_model
+
+    # Check if the device is specified twice. If true then the device parameter
+    # of model handler is overridden.
+    self._deduplicate_device_value(device)
     _validate_constructor_args_hf_pipeline(self._task, self._model)
 
+  def _deduplicate_device_value(self, device: str):
```

Comment:
Would be good to throw if device != CPU or GPU here as well. Also would be good to convert to uppercase if device is not None